src/main/java/banking/BankingSystem.java

```diff
@@ -0,0 +1,171 @@
+package banking;
+
+import banking.dao.AccountDAO;
+import banking.model.Account;
+import banking.model.Card;
+
+import java.util.Scanner;
+
+import static java.lang.System.exit;
+
+public class BankingSystem {
+
+    private Account currentAccount;
+    private int amountSend;
+    private boolean isLogin;
+    private final CardGenerator cardGenerator;
+    private final CardValidator cardValidator;
+    private final AccountDAO accountDAO;
+    private final Scanner scanner;
+
+    public BankingSystem(AccountDAO accountDAO) {
+        cardGenerator = new CardGenerator();
+        cardValidator = new CardValidator();
+        this.accountDAO = accountDAO;
+        scanner = new Scanner(System.in);
+    }
+
+    public void showMenu() {
+        while (true) {
+            showMenuInfo();
+            chooseMenu();
+        }
+    }
+
+    public void showMenuInfo() {
+        if (isLogin) {
+            System.out.println("""
+                    1. Balance
+                    2. Add income
+                    3. Do transfer
+                    4. Close account
+                    5. Log out
+                    0. Exit""");
+        } else {
+            System.out.println("""
+                    1. Create an account
+                    2. Log into account
+                    0. Exit""");
+        }
+    }
+
+    public void chooseMenu() {
+        int choice = scanner.nextInt();
+        if (isLogin) {
+            loggedMenu(choice);
+        } else {
+            registerMenu(choice);
+        }
+    }
+
+    private void registerMenu(int choose) {
+        switch (choose) {
+            case 1 -> createAccount();
+            case 2 -> logInOption();
+            case 0 -> {
+                scanner.close();
+                System.out.println("Bye!");
+                exit(0);
```

Comment:
You have duplication in the `loggedMenu` method, can you move this piece of code to a common method?

Reply:
fixed